### PR TITLE
8345684: OperatingSystemMXBean.getSystemCpuLoad() throws NPE

### PR DESCRIPTION
**Pull request by fmeum**
The return value of Metrics#getCpuSetCpus may change over time, including from non-null to null across the two calls in this method.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8345684](https://bugs.openjdk.org/browse/JDK-8345684): OperatingSystemMXBean.getSystemCpuLoad() throws NPE (**Bug** - P4)

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Kevin Walls](https://openjdk.org/census#kevinw) (@kevinjwalls - **Reviewer**)
 * [Severin Gehwolf](https://openjdk.org/census#sgehwolf) (@jerboaa - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22611/head:pull/22611` \
`$ git checkout pull/22611`

Update a local copy of the PR: \
`$ git checkout pull/22611` \
`$ git pull https://git.openjdk.org/jdk.git pull/22611/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22611`

View PR using the GUI difftool: \
`$ git pr show -t 22611`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22611.diff">https://git.openjdk.org/jdk/pull/22611.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22611#issuecomment-2523778134)
</details>

### CODE REVIEW

File: src/jdk.management/unix/classes/com/sun/management/internal/OperatingSystemImpl.java
Comment (RogerRiggs) [inline src/jdk.management/unix/classes/com/sun/management/internal/OperatingSystemImpl.java:278]: Might be worth a comment about the volatility of getCpuSetCpus() to head off someone changing it back.

Comment (fmeum) [reply src/jdk.management/unix/classes/com/sun/management/internal/OperatingSystemImpl.java:280]: I added a comment explaining both the volatility and the potential cost.

**Review by dholmes-ora — APPROVED** [review]
LGTM.

Aside: from a Java API design perspective I think these container API's should be returning empty arrays not null

**Review by kevinjwalls — APPROVED** [review]
Looks good, thanks 8-)

**Comment by kevinjwalls** [comment]
Hi, 
This looks fine, I'm sure it's ok to proceed without waiting for a re-review from Roger about the comment.

It would be good to see the "Checks" tab show more than just jcheck.
Looks like this needs to be enabled in the personal fork.
https://wiki.openjdk.org/display/SKARA/Testing

Then we get some basic builds and tests done for us, and can go ahead with (even more) confidence!

**Comment by kevinjwalls** [comment]
That's great.  As the sponsor label has gone, when these few tests finish I think you'll need to say /integrate again then I can sponsor.

**Review by jerboaa — APPROVED** [review]
LGTM

**Comment by kevinjwalls** [comment]
I think we can consider that failure some annoying noise.  It's not expected, but this Java-only change isn't causing it, other builds are fine, we should proceed.

**Comment by fmeum** [comment]
@kevinjwalls Thanks! Apologies in advance if this is not the right process, but could this be cherry-picked into JDK 21?

**Comment by fmeum** [comment]
@jerboaa It looks like I would need to be at least an *Author* to follow these instructions. Since this is my second merged PR to the OpenJDK, could I become one?

**Comment by jerboaa** [comment]
> @jerboaa It looks like I would need to be at least an _Author_ to follow these instructions. Since this is my second merged PR to the OpenJDK, could I become one?

For the skara bot command, yes. For manual contribution, no. See https://wiki.openjdk.org/display/JDKUpdates/How+to+contribute+or+backport+a+fix

**Comment by jerboaa** [comment]
> Since this is my second merged PR to the OpenJDK, could I become one?

If you want to be, the process for becoming OpenJDK author is here:
https://openjdk.org/projects/#project-author

The relevant project is `jdk`:
https://openjdk.org/census#jdk

**Pull request closed without merging**
